jenkins/e2e-image/e2e-runner.sh

```diff
@@ -181,19 +181,29 @@ function install_google_cloud_sdk_tarball() {
     gcloud info
 }
 
-# Figures out the builtin k8s version of a GCI image.
+# Sets release vars using GCI image builtin k8s version.
+# If JENKINS_GCI_PATCH_K8S is set, uses the latest CI build on the same branch
+# instead.
 # Assumes: JENKINS_GCI_HEAD_IMAGE_FAMILY and KUBE_GCE_MASTER_IMAGE
-function get_gci_k8s_version() {
+function set_release_vars_from_gci_builtin_version() {
     if ! [[ "${JENKINS_USE_GCI_VERSION:-}" =~ ^[yY]$ ]]; then
         echo "JENKINS_USE_GCI_VERSION must be set."
         exit 1
     fi
     if [[ -z "${JENKINS_GCI_HEAD_IMAGE_FAMILY:-}" ]] || [[ -z "${KUBE_GCE_MASTER_IMAGE:-}" ]]; then
-        echo "JENKINS_GCI_HEAD_IMAGE_FAMILY and KUBE_GCE_MASTER_IMAGE must be set."
+        echo "JENKINS_GCI_HEAD_IMAGE_FAMILY and KUBE_GCE_MASTER_IMAGE must both be set."
         exit 1
     fi
-    local -r gci_k8s_version="v$(gsutil cat gs://container-vm-image-staging/k8s-version-map/${KUBE_GCE_MASTER_IMAGE})"
-    echo "${gci_k8s_version}"
+    local -r gci_k8s_version="$(gsutil cat gs://container-vm-image-staging/k8s-version-map/${KUBE_GCE_MASTER_IMAGE})"
+    if [[ "${JENKINS_GCI_PATCH_K8S}" =~ ^[yY]$ ]]; then
+      # We always want to test against the builtin k8s version, but occationally
```

Comment:
nit: I think this function is using 4-space indents? doesn't really matter though, this isn't python.